remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/mappers/ActionAdapter.kt

```diff
@@ -47,14 +48,16 @@ class ActionAdapter constructor(
             OldActionType.DISMISS.toString() -> JsonActionType.DISMISS.jsonValue
             OldActionType.APP_TP_ONBOARDING.toString() -> JsonActionType.APP_TP_ONBOARDING.jsonValue
             OldActionType.PLAYSTORE.toString() -> JsonActionType.PLAYSTORE.jsonValue
-            OldActionType.URL.toString() -> JsonActionType.URL.jsonValue
+            OldActionType.URL.toString() -> JsonActionType.SHARE.jsonValue
+            OldActionType.SHARE.toString() -> JsonActionType.URL.jsonValue
             else -> actionType
         }
     }
 
     class ActionJson(
         val actionType: String,
         val value: String,
+        val additionalParameters: JsonAdditionalParameters? = null,
```

Comment:
Ditto, why the specific `JsonAdditionalParameters` type and not just a `String`?
Or better, cannot that be part of `value` directly?

Reply:
Some actions may receive extra data. Think about that as an optional payload.

We can't change `value` format or expectations, as we want to keep consistent across all actions and it will break backwards compatibility. `additionalParameters` should be instead a map to be generic.

e.g: Share action will be used to share `value` on a share sheet. it can receive extra values if we want to customize title or other attributes of that share sheet.

Reply:
Make sense. 
I would not make it a custom type though. Either a `String` or a `Map<something, something>`

Reply:
Changed to `Map<String, String>` 👍